test/python/visualization/test_circuit_drawer.py

```diff
@@ -119,3 +120,13 @@ def test_wire_order_raises(self):
 
         with self.assertWarnsRegex(RuntimeWarning, "Cregbundle set"):
             visualization.circuit_drawer(circuit, cregbundle=True, wire_order=[0, 1, 2, 5, 4, 3])
+
+    def test_reverse_bits(self):
+        """Test reverse_bits should not raise warnings when no classical qubits:
+        See: https://github.com/Qiskit/qiskit-terra/pull/8689"""
+        circuit = QuantumCircuit(3)
+        circuit.x(1)
+
+        with catch_warnings():
+            simplefilter("error")
+            visualization.circuit_drawer(circuit, reverse_bits=True)
```

Comment:
It would be good to test the output of this against the expected one as well, just to be sure.

I think the test suite should automatically fail if the warning is emitted without messing with `catch_warnings`?

Reply:
> It would be good to test the output of this against the expected one as well, just to be sure.

checking the result in 6a1e2234cfc957a66a944c213522550de2ce1716

> I think the test suite should automatically fail if the warning is emitted without messing with `catch_warnings`?

I dont disagree, but I  think we have some warnings around. Maybe for another PR?

Reply:
I removed the warning missing